relay-server/src/service.rs

```diff
@@ -32,85 +32,41 @@ use crate::{endpoints, utils};
 
 pub static REGISTRY: OnceBox<Registry> = OnceBox::new();
 
-/// Common error type for the relay server.
-#[derive(Debug)]
-pub struct ServerError {
-    inner: Context<ServerErrorKind>,
-}
-
 /// Indicates the type of failure of the server.
-#[derive(Debug, Fail, Copy, Clone, PartialEq, Eq, Hash)]
-pub enum ServerErrorKind {
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, thiserror::Error)]
+pub enum ServerError {
```

Comment:
The `ServerError` type is now replaced by an opaque `anyhow::Error`, and the former `kind` just there to create the context message. I chose to rename it for brevity.

An alternative to this design is that we inline the error messages directly like so:

```rust
  bind(4711).context("bind to interface failed")?
```